src/main/java/org/jenkins_ci/plugins/flexible_publish/FlexiblePublisher.java

```diff
@@ -111,10 +112,18 @@ public boolean prebuild(final AbstractBuild<?, ?> build, final BuildListener lis
 
     @Override
     public boolean perform(final AbstractBuild<?, ?> build, final Launcher launcher, final BuildListener listener)
-                                                                                                throws InterruptedException, IOException {
-        for (ConditionalPublisher publisher : publishers)
-            if (!publisher.perform(build, launcher, listener))
+            throws InterruptedException, IOException {
+        for (ConditionalPublisher publisher : publishers) {
+            try {
+                if (!publisher.perform(build, launcher, listener)) {
+                    setResult(build, Result.FAILURE);
+                    listener.getLogger().println("[flexible-publish] ConditionalPublisher set build result to FAILURE.");
```

Comment:
Here we just log that ConditionalPublisher set failure.
The real error reason must be printed from plugin itself.